sdk/go/pulumi/policyx/policy.go

```diff
@@ -66,6 +66,8 @@ type ResourceValidationArgs struct {
 	Resource AnalyzerResource
 	// Config is the policy configuration.
 	Config map[string]any
+	// DryRun indicates if the current operation is a dry run (preview).
```

Comment:
This isn't ideal, gonna change this in a later PR but for a quick fix to get this dry run test passing for Go it suffices.